multispecies_whale_detection/front_end.py

```diff
@@ -181,7 +217,30 @@ def call(self, waveform, training=False):
     num_frequency_bins = magnitude.shape[-1]
     hz_per_bin = sample_rate / 2 / num_frequency_bins
 
-    db = amplitude_ratio_to_db(magnitude + self._config.log_stabilizer)
+    frequency_scaling = self._config.frequency_scaling
+    if not frequency_scaling:
+      db = amplitude_ratio_to_db(magnitude + self._config.log_stabilizer)
+    elif isinstance(frequency_scaling, MelScalingConfig):
+      mel_scale_power = tf.matmul(
```

Comment:
Done, although I was on the fence about this. Arguably linear_to_mel_weight_matrix was the helper function.

(I sometimes get discouraged about extracting methods because of linter docstring requirements. In defense of the linter, too short and too absent docstrings are also trouble.)